### PR TITLE
shell_(parse|split|escape): fix some issues with deprecation

### PR DESCRIPTION
**Pull request by StefanKarpinski**
* shell_parse needed a special keyword arg but didn't have one
* seems better to have shell_split and shell_escape default to _not_ having any special shell characters
* instead the usages of shell_parse in the implementation of the backticks syntax supplies the now-special shell characters and the corresponding use of shell_escape when displaying them too

### CODE REVIEW

**Comment by StefanKarpinski** [comment]
See https://github.com/JuliaLang/julia/pull/19786. cc: @amitmurthy, @tkelman. This addresses your concerns from that PR by making `shell_split` and `shell_escape` default to their old behavior. The usages specifically for `Cmd` objects and syntax do `special = Base.shell_special`.

File: base/shell.jl [hunk 1]
```diff
@@ -158,30 +166,30 @@ end
...
 """
-    shell_escape(args::Union{Cmd,AbstractString...}; special::AbstractString="$shell_special")
+    shell_escape(args::Union{Cmd,AbstractString...}; special::AbstractString="")
 
 The unexported `shell_escape` function is the inverse of the unexported `shell_split` function:
 it takes a string or command object and escapes any special characters in such a way that calling
 `shell_split` on it would give back the array of words in the original command. The `special`
 keyword argument controls what characters in addition to whitespace, backslashes, quotes and
-dollar signs are considered to be special. Examples:
+dollar signs are considered to be special (default: none). Examples:
```
Comment (tkelman) [inline base/shell.jl]: "in addition" isn't accurate any more then?

File: base/shell.jl [hunk 2]
```diff
@@ -4,7 +4,15 @@
 
 const shell_special = "#{}()[]<>|&*?~;"
 
-function shell_parse(str::AbstractString, interpolate::Bool=true)
+ # needs to be factored out so depwarn only warns once
+ @noinline warn_shell_special(special) =
```
Comment (tkelman) [inline base/shell.jl]: why the leading spaces?

Comment (StefanKarpinski) [inline base/shell.jl]: editor playing tricks on me?

File: base/shell.jl (hunk 1 above)
Comment (StefanKarpinski) [inline base/shell.jl]: It is: whitespace, backslashes, quotes and dollar signs are always considered special; the `special` keyword allows you to supply a string of characters that are *also* considered special.

File: base/shell.jl [hunk 3]
```diff
@@ -158,30 +166,30 @@ end
...
 `shell_split` on it would give back the array of words in the original command. The `special`
 keyword argument controls what characters in addition to whitespace, backslashes, quotes and
-dollar signs are considered to be special. Examples:
+dollar signs are considered to be special (default: none). Examples:
+
+    julia> Base.shell_escape("cat", "/foo/bar baz", "&&", "echo", "done")
+    "cat '/foo/bar baz' && echo done"
 
     julia> Base.shell_escape("echo", "this", "&&", "that")
     "echo this '&&' that"
```
Comment (tkelman) [inline base/shell.jl]: no longer displays with quotes around the `&&`, and these should be doctests so this would have been noticed

Comment (StefanKarpinski) [inline base/shell.jl]: Please make a PR to fix it.

Comment (tkelman) [inline base/shell.jl]: At the cost of both 0.5.2 and 0.6.0-rc1 coming out later because I'd be spending time fixing this instead, sure.

**Pull request merged**